tests/baselines/reference/strictOptionalProperties1.errors.txt

```diff
@@ -1,44 +1,55 @@
+    Type 'undefined' is not assignable to type 'string'.
+tests/cases/compiler/strictOptionalProperties1.ts(105,7): error TS2375: Type '[number, undefined]' is not assignable to type '[number, string?, boolean?]' with exactOptionalPropertyTypes: true. Consider adding 'undefined' to the types of the target's properties.
+  Type at position 1 in source is not compatible with type at position 1 in target.
+    Type 'undefined' is not assignable to type 'string'.
+tests/cases/compiler/strictOptionalProperties1.ts(106,7): error TS2375: Type '[number, string, undefined]' is not assignable to type '[number, string?, boolean?]' with exactOptionalPropertyTypes: true. Consider adding 'undefined' to the types of the target's properties.
+  Type at position 2 in source is not compatible with type at position 2 in target.
+    Type 'undefined' is not assignable to type 'boolean'.
+tests/cases/compiler/strictOptionalProperties1.ts(107,7): error TS2375: Type '[number, undefined, undefined]' is not assignable to type '[number, string?, boolean?]' with exactOptionalPropertyTypes: true. Consider adding 'undefined' to the types of the target's properties.
+  Type at position 1 in source is not compatible with type at position 1 in target.
+    Type 'undefined' is not assignable to type 'string'.
+tests/cases/compiler/strictOptionalProperties1.ts(111,7): error TS2375: Type '{ foo: undefined; }' is not assignable to type '{ foo?: number; }' with exactOptionalPropertyTypes: true. Consider adding 'undefined' to the types of the target's properties.
+  Types of property 'foo' are incompatible.
+    Type 'undefined' is not assignable to type 'number'.
 tests/cases/compiler/strictOptionalProperties1.ts(119,5): error TS2411: Property 'bar' of type 'string | undefined' is not assignable to 'string' index type 'string'.
 
 
-==== tests/cases/compiler/strictOptionalProperties1.ts (17 errors) ====
+==== tests/cases/compiler/strictOptionalProperties1.ts (16 errors) ====
     function f1(obj: { a?: string, b?: string | undefined }) {
         let a = obj.a;  // string | undefined
         let b = obj.b;  // string | undefined
         obj.a = 'hello';
         obj.b = 'hello';
         obj.a = undefined;  // Error
         ~~~~~
-!!! error TS2322: Type 'undefined' is not assignable to type 'string'.
+!!! error TS2412: Type 'undefined' is not assignable to type 'string' with exactOptionalPropertyTypes: true. Consider adding 'undefined' to the type of the target.
         obj.b = undefined;
     }
     
     function f2(obj: { a?: string, b?: string | undefined }) {
         obj = obj;
         obj.a = obj.a;  // Error
         ~~~~~
-!!! error TS2322: Type 'string | undefined' is not assignable to type 'string'.
-!!! error TS2322:   Type 'undefined' is not assignable to type 'string'.
+!!! error TS2412: Type 'string | undefined' is not assignable to type 'string' with exactOptionalPropertyTypes: true. Consider adding 'undefined' to the type of the target.

```

Comment:
this error is not obvious, but self-assignments of optional properties require adding `undefined`